crates/biome_js_analyze/src/analyzers/nursery/no_duplicate_else_if.rs

```diff
@@ -0,0 +1,220 @@
+                                    is_subset(current_or_operand, or_operand)
+                                })
+                            })
+                            .cloned()
+                            .collect()
+                    })
+                    .collect();
+                if list_to_check.iter().any(|f| f.is_empty()) {
+                    return node.clone().test().ok();
+                }
+            }
+        }
+
+        None
+    }
+
+    fn diagnostic(
+        _: &RuleContext<Self>,
+        expression_inside_if: &Self::State,
+    ) -> Option<RuleDiagnostic> {
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            expression_inside_if.range(),
+            markup! {
+                "This branch can never execute. Its condition is a duplicate or covered by previous conditions in the if-else-if chain."
+            },
+        ))
+    }
+}
+
+fn flatten_parent(node: &AnyJsExpression) -> AnyJsExpression {
+    let expression_inside_parent = node
+        .as_js_parenthesized_expression()
+        .and_then(|parent| parent.expression().ok());
+
+    if let Some(expression) = expression_inside_parent {
+        return expression;
+    }
+    node.clone()
+}
+fn split_by_logical_operator_wrapper(
+    operator: JsLogicalOperator,
+    node: &AnyJsExpression,
+) -> Vec<AnyJsExpression> {
+    let mut result: Vec<AnyJsExpression> = vec![];
+    split_by_logical_operator(operator, node, &mut result);
+    result
+}
+fn split_by_logical_operator(
+    operator: JsLogicalOperator,
+    node: &AnyJsExpression,
+    result: &mut Vec<AnyJsExpression>,
+) {
+    match node {
+        AnyJsExpression::JsLogicalExpression(logic_expression) => {
+            if let Ok(operator_token) = logic_expression.operator() {
+                if operator_token != operator {
+                    result.push(node.clone());
+                    return;
+                }
+            }
+            if let (Ok(left_node), Ok(right_node)) =
+                (logic_expression.left(), logic_expression.right())
+            {
+                split_by_logical_operator(operator, &flatten_parent(&left_node), result);
+                split_by_logical_operator(operator, &flatten_parent(&right_node), result);
+            } else {
+                result.push(node.clone());
+            }
+        }
+        _ => {
+            result.push(node.clone());
+        }
+    }
+}
+
+fn equal(a: &AnyJsExpression, b: &AnyJsExpression) -> bool {
+    if a.syntax().kind() != b.syntax().kind() {
+        return false;
+    }
+    if a.syntax().kind() == JsSyntaxKind::JS_LOGICAL_EXPRESSION {
+        if let (Some(a_exp), Some(b_exp)) =
+            (a.as_js_logical_expression(), b.as_js_logical_expression())
+        {
+            if a_exp.operator() == b_exp.operator()
+                && (a_exp.operator() == Ok(JsLogicalOperator::LogicalAnd)
+                    || a_exp.operator() == Ok(JsLogicalOperator::LogicalOr))
```

Comment:
You could use `matches!`
```suggestion
                && matches!(a_exp.operator(), Ok(JsLogicalOperator::LogicalAnd | JsLogicalOperator::LogicalO))
```